common/kernel/sdc.cc

```diff
@@ -0,0 +1,453 @@
+                } else {
+                    log_error("unsupported argument '%s' to create_clock\n", s.c_str());
+                }
+            } else {
+                for (const auto &ety : arg.list) {
+                    NetInfo *net = nullptr;
+                    if (ety.type == TclEntity::ENTITY_PIN)
+                        net = ety.get_net(ctx);
+                    else if (ety.type == TclEntity::ENTITY_NET)
+                        net = ctx->nets.at(ety.name).get();
+                    else if (ety.type == TclEntity::ENTITY_PORT)
+                        net = ctx->ports.at(ety.name).net;
+                    else
+                        log_error("create_clock applies only to cells, cell pins, or IO ports (line %d)\n", lineno);
+
+                    ctx->addClock(net->name, 1000.0f / period);
+                }
+            }
+        }
+        return std::string{};
+    }
+
+    TclValue cmd_set_false_path(const std::vector<TclValue> &arguments)
+    {
+        NetInfo *from = nullptr;
+        NetInfo *to = nullptr;
+
+        for (int i = 1; i < int(arguments.size()); i++) {
+            auto &arg = arguments.at(i);
+            if (arg.is_string) {
+                std::string s = arg.str;
+
+                bool is_from = true;
+                if (s == "-to") {
+                    is_from = false;
+                } else if (s != "-from") {
+                    log_error("expecting either -to or -from to set_false_path(line %d)\n", lineno);
+                }
+
+                i++;
+                auto &val = arguments.at(i);
+                if (val.is_string) {
+                    log_error("expecting TclValue argument to -from (line %d)\n", lineno);
+                }
+
+                if (val.list.size() != 1) {
+                    log_error("Expected a single TclEntity as argument to -to/-from (line %d)\n", lineno);
+                }
+
+                auto &ety = val.list.at(0);
+
+                NetInfo *net = nullptr;
+                if (ety.type == TclEntity::ENTITY_PIN)
+                    net = ety.get_net(ctx);
+                else if (ety.type == TclEntity::ENTITY_NET)
+                    net = ctx->nets.at(ety.name).get();
+                else if (ety.type == TclEntity::ENTITY_PORT)
+                    net = ctx->ports.at(ety.name).net;
+                else
+                    log_error("set_false_path applies only to nets, cell pins, or IO ports (line %d)\n", lineno);
+
+                if (is_from) {
+                    from = net;
+                } else {
+                    to = net;
+                }
+            }
+        }
+
+        if (from == nullptr) {
+            log_error("-from is required for set_false_path (line %d)\n", lineno);
+        } else if (to == nullptr) {
+            log_error("-to is required for set_false_path (line %d)\n", lineno);
+        }
+
+        log_info("set_false_path from: %s, to: %s\n", from->name.c_str(ctx), to->name.c_str(ctx));
```

Comment:
this should probably be a warning that the `set_false_path` is currently being ignored

Reply:
Fixed